includes/admin/settings/SameDayDeliveryService.php

```diff
@@ -0,0 +1,124 @@
+
+    /**
+     * @param  string $carrierName
+     */
+    public function __construct(string $carrierName)
+    {
+        $this->settingsCollection = WCMYPA()->setting_collection->where('carrier', $carrierName);
+    }
+
+    /**
+     * @return bool
+     */
+    public function shouldShowSameDayDelivery(): bool
+    {
+        $carrierIsActive     = (bool) $this->settingsCollection->getByName(WCMYPA_Settings::SETTING_CARRIER_DELIVERY_ENABLED);
+        $sameDayFromSettings = (bool) $this->settingsCollection->getByName(
+            WCMYPA_Settings::SETTING_CARRIER_DEFAULT_EXPORT_SAME_DAY_DELIVERY
+        );
+
+        if (! $carrierIsActive || ! $sameDayFromSettings) {
+            return false;
+        }
+
+        $date = (new DateTime())->setTimezone(new DateTimeZone('Europe/Amsterdam'));
+        $now  = $date->getTimestamp() + $date->getOffset();
+
+        return $this->isInSameDayTimeSlot($now) && ! $this->hasDropOffDelay() && $this->isDropOffPossible($now);
+    }
+
+    /**
+     * @param  int $now
+     *
+     * @return bool
+     */
+    private function isInSameDayTimeSlot(int $now): bool
+    {
+        $sameDayCutoffTimeFromSettings   = $this->settingsCollection->getByName(
+            WCMYPA_Settings::SETTING_CARRIER_DEFAULT_EXPORT_SAME_DAY_DELIVERY_CUTOFF_TIME
+        );
+        $cutOffTimeFromSettings          = $this->settingsCollection->getByName(
+            WCMYPA_Settings::SETTING_CARRIER_CUTOFF_TIME
+        );
+        $sameDayCutOffBeforeNormalCutOff = strtotime($sameDayCutoffTimeFromSettings) < strtotime(
+                $cutOffTimeFromSettings
+            );
+
+        if (! $sameDayCutOffBeforeNormalCutOff) {
+            return false;
+        }
+
+        $beforeSameDayCutOffTime = $now < strtotime($sameDayCutoffTimeFromSettings);
+        $afterRegularCutOffTime  = strtotime($cutOffTimeFromSettings) < $now;
+
+        return $beforeSameDayCutOffTime || $afterRegularCutOffTime;
+    }
+
+    /**
+     * @param  int $now
+     *
+     * @return bool
+     */
+    private function isDropOffPossible(int $now): bool
+    {
+        $cutOffTime = $this->settingsCollection->getByName(WCMYPA_Settings::SETTING_CARRIER_CUTOFF_TIME);
+        $dropOffDays = $this->settingsCollection->getByName(WCMYPA_Settings::SETTING_CARRIER_DROP_OFF_DAYS);
+
+        if ($now > $cutOffTime) {
+            return $this->isDropOffTomorrowPossible($dropOffDays);
+        }
+
+        return $this->isDropOffTodayPossible($dropOffDays);
+    }
+
+    /**
+     * @param  array $dropOffDays
+     *
+     * @return bool
+     */
+    private function isDropOffTodayPossible(array $dropOffDays): bool
+    {
+        return in_array((date('N')), $dropOffDays, true);
+    }
+
+    /**
+     * @param  array $dropOffDays
+     *
+     * @return bool
+     */
+    private function isDropOffTomorrowPossible(array $dropOffDays): bool
+    {
+        return in_array((string) (date('N') + 1), $dropOffDays, true);
```

Comment:
Misschien is `new DateTime('tomorrow')->format('N');` wat makkelijker te lezen. Je kunt ook nog een constante maken van N als je dat "leuk" vindt, zodat het bij het lezen direct duidelijk is dat je een weekdag ophaalt ofzo. 

Het wordt wat langer maar bij wat er nu staat moest ik helemaal opzoeken wat het precies doet, php heeft nogal wat obscure date dingen :)